connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogTableSpec.scala

```diff
@@ -0,0 +1,343 @@
+package com.datastax.spark.connector.datasource
+
+import com.datastax.oss.driver.api.core.CqlIdentifier.fromInternal
+import org.apache.spark.SparkException
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.analysis.{NoSuchNamespaceException, NoSuchTableException, TableAlreadyExistsException}
+
+import scala.collection.JavaConverters._
+
+class CassandraCatalogTableSpec extends CassandraCatalogSpecBase {
+
+  "A Cassandra Catalog Table Support" should "initialize successfully" in {
+    spark.sessionState.catalogManager.currentCatalog.name() should include("Catalog cassandra")
+  }
+
+  val testTable = "testTable"
+
+  it should "create a table with a partition key" in {
+    createDefaultKs()
+    spark.sql(s"CREATE TABLE $defaultKs.$testTable (key Int, value STRING) USING cassandra PARTITIONED BY (key)")
+    getTable(defaultKs, testTable).getPartitionKey.get(0).getName.asInternal() should be ("key")
```

Comment:
Do we need/should we have an explicit wait for table creation here and/or a metadata sync before trying to check results from the metadata?  Same question for other tests within this spec I suppose.

Reply:
I mentioned this a bit previously with Jarek. Since all catalog sessions share the same underlying session (pool) they all have the same metadata access as well. Which means that any statements which do DDL will block until the metadata is updated internally. So we really don't have to, I added in a bunch of these waits because I wanted to verify that things existed in some places but I don't think they are really necessary.